# EDA Titulos

In [36]:
import pandas as pd
import numpy as np

df_titulos = pd.read_csv("datos/netflix_titles.csv", index_col=0)
df_originals = pd.read_csv("datos/netflix_originals.csv", index_col=0)
pd.set_option("display.max_columns", None)
df_titulos.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

# Infomación

In [37]:
df_titulos.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

In [38]:
df_titulos.shape

(8807, 12)

In [39]:
df_titulos["show_id"].value_counts()

show_id
s8807    1
s1       1
s2       1
s3       1
s4       1
        ..
s11      1
s12      1
s13      1
s14      1
s15      1
Name: count, Length: 8807, dtype: int64

In [40]:
df_titulos.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration        4813
listed_in          0
description        0
dtype: int64

In [41]:
df_titulos.duplicated(subset="title").sum()

#Hay titulos duplicados

np.int64(0)

In [42]:
df_titulos["type"].value_counts()

#Hay 6131 Movie y 2676 TV Show

type
Movie      6131
TV Show    2676
Name: count, dtype: int64

In [43]:
df_titulos.describe(include=object).T

count unique                                                top  \
show_id      8807   8807                                              s8807   
type         8807      2                                              Movie   
title        8807   8807                                             Zubaan   
director     6173   4528                                      Rajiv Chilaka   
cast         7982   7692                                 David Attenborough   
country      7976    748                                      United States   
date_added   8797   1767                                    January 1, 2020   
rating       8803     17                                              TV-MA   
duration     3994    185                                          2 Seasons   
listed_in    8807    514                       Dramas, International Movies   
description  8807   8775  Paranormal activity at a lush, abandoned prope...   

             freq  
show_id         1  
type         6131  
title           1  
director       19  
cast           19  
country      2818  
date_added    109  
rating       3207  
duration      425  
listed_in     362  
description     4

In [44]:
np.round(df_titulos.loc[:,["rating", "type"]].value_counts(normalize=True)*100, 2)

rating    type   
TV-MA     Movie      23.42
TV-14     Movie      16.21
TV-MA     TV Show    13.01
R         Movie       9.05
TV-14     TV Show     8.33
TV-PG     Movie       6.13
PG-13     Movie       5.57
TV-PG     TV Show     3.67
PG        Movie       3.26
TV-Y7     TV Show     2.22
TV-Y      TV Show     2.00
TV-Y7     Movie       1.58
TV-Y      Movie       1.49
TV-G      Movie       1.43
          TV Show     1.07
NR        Movie       0.85
G         Movie       0.47
NR        TV Show     0.06
TV-Y7-FV  Movie       0.06
UR        Movie       0.03
NC-17     Movie       0.03
R         TV Show     0.02
84 min    Movie       0.01
74 min    Movie       0.01
66 min    Movie       0.01
TV-Y7-FV  TV Show     0.01
Name: proportion, dtype: float64

In [45]:
df_titulos[["rating","type"]].value_counts()

rating    type   
TV-MA     Movie      2062
TV-14     Movie      1427
TV-MA     TV Show    1145
R         Movie       797
TV-14     TV Show     733
TV-PG     Movie       540
PG-13     Movie       490
TV-PG     TV Show     323
PG        Movie       287
TV-Y7     TV Show     195
TV-Y      TV Show     176
TV-Y7     Movie       139
TV-Y      Movie       131
TV-G      Movie       126
          TV Show      94
NR        Movie        75
G         Movie        41
NR        TV Show       5
TV-Y7-FV  Movie         5
UR        Movie         3
NC-17     Movie         3
R         TV Show       2
84 min    Movie         1
74 min    Movie         1
66 min    Movie         1
TV-Y7-FV  TV Show       1
Name: count, dtype: int64

In [46]:
df_titulos.loc[:,["director","type"]].value_counts()

director                  type 
Rajiv Chilaka             Movie    19
Raúl Campos, Jan Suter    Movie    18
Suhas Kadav               Movie    16
Marcus Raboy              Movie    15
Jay Karas                 Movie    14
                                   ..
Çagan Irmak               Movie     1
Ísold Uggadóttir          Movie     1
Óskar Thór Axelsson       Movie     1
Billy Corben              Movie     1
Rajiv Chilaka, Owll Mina  Movie     1
Name: count, Length: 4576, dtype: int64

In [137]:
df_titulos.loc[:,["director","type"]].value_counts(normalize=True)

director                  type 
Rajiv Chilaka             Movie    0.003078
Raúl Campos, Jan Suter    Movie    0.002916
Suhas Kadav               Movie    0.002592
Marcus Raboy              Movie    0.002430
Jay Karas                 Movie    0.002268
                                     ...   
Çagan Irmak               Movie    0.000162
Ísold Uggadóttir          Movie    0.000162
Óskar Thór Axelsson       Movie    0.000162
Billy Corben              Movie    0.000162
Rajiv Chilaka, Owll Mina  Movie    0.000162
Name: proportion, Length: 4576, dtype: float64

In [133]:
df_titulos.loc[:,"country"].value_counts()

country
United States                                    2818
India                                             972
United Kingdom                                    419
Japan                                             245
South Korea                                       199
                                                 ... 
Russia, Spain                                       1
Croatia, Slovenia, Serbia, Montenegro               1
Japan, Canada                                       1
United States, France, South Korea, Indonesia       1
United Arab Emirates, Jordan                        1
Name: count, Length: 748, dtype: int64

In [134]:
df_titulos.loc[:,"country"].value_counts(normalize=True)

country
United States                                    0.353310
India                                            0.121866
United Kingdom                                   0.052533
Japan                                            0.030717
South Korea                                      0.024950
                                                   ...   
Russia, Spain                                    0.000125
Croatia, Slovenia, Serbia, Montenegro            0.000125
Japan, Canada                                    0.000125
United States, France, South Korea, Indonesia    0.000125
United Arab Emirates, Jordan                     0.000125
Name: proportion, Length: 748, dtype: float64

# Conclusiones generales

En este dataframe hay titulos de peliculas y series de Netflix.

Existen 13 columnas y 8807 peliculas y series. La mayoria son de tipo objeto. Existen 11 columnas de tipo objeto y 2 int.

Hay valores nulos pero no duplicados. Apuntar columnas de nulos

Exiten 6131 peliculas y 2676 series.

En cuanto a las clasificaciones de edad la mayoria de peliculas estan clasificadas con TV-MA. Son 2062 filas que equivale a un 23.42% del total. La clasficación de edad de las series que más porcentaje tiene es TV-MA con un total de 1145 que equivale a un 13.01% del total. Dentro de esta categoría existen datos, en concreto 3 filas, que nos generarán inconsistencias al hacer un analisís en el futuro, estos datos tienen que ver con la duración de las peliculas.

El director que más peliculas o tv shows es Rajiv Chilaka con 19 pelliculas.

En un futuro se podría poner como indice la columna 'show_id'. Resulta ser el identificador de cada serie y son valores unicos.

El pais con más producciones es United States con 2818 que equivale a un 35.33% del total.

In [121]:
df_concat = df_titulos.merge(df_originals, how="inner", right_on="Title", left_on="title")
df_concat.head()

show_id   type                 title             director  \
0      s1  Movie  Dick Johnson Is Dead      Kirsten Johnson   
1    s142  Movie            Extraction     Steven C. Miller   
2    s625  Movie            Prime Time         Jakub Piątek   
3    s835  Movie          Blue Miracle       Julio Quintana   
4    s837  Movie             Ghost Lab  Paween Purijitpanya   

                                                cast  \
0                                                NaN   
1  Bruce Willis, Kellan Lutz, Gina Carano, D.B. S...   
2  Bartosz Bielenia, Magdalena Popławska, Andrzej...   
3  Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...   
4  Thanapob Leeratanakachorn, Paris Intarakomalya...   

                                 country          date_added  release_year  \
0                          United States  September 25, 2021          2020   
1  United States, United Kingdom, Canada   September 1, 2021          2015   
2                                    NaN       June 30, 2021          2021   
3                          United States        May 27, 2021          2021   
4                               Thailand        May 27, 2021          2021   

  rating duration                                          listed_in  \
0  PG-13   90 min                                      Documentaries   
1      R   82 min                                 Action & Adventure   
2  TV-MA      NaN            Dramas, International Movies, Thrillers   
3  TV-PG      NaN  Children & Family Movies, Dramas, Faith & Spir...   
4  TV-MA  118 min     Horror Movies, International Movies, Thrillers   

                                         description                 Title  \
0  As her father nears the end of his life, filmm...  Dick Johnson Is Dead   
1  When a retired CIA agent is kidnapped, his son...            Extraction   
2  On New Year’s Eve 1999, an armed man enters a ...            Prime Time   
3  To save their cash-strapped orphanage, a guard...          Blue Miracle   
4  After witnessing a haunting in their hospital,...             Ghost Lab   

         Genre         Premiere  Runtime  IMDB Score Language  
0  Documentary  October 2, 2020       90         7.5  English  
1       Action   April 24, 2020      117         6.7  English  
2     Thriller   April 14, 2021       91         5.7   Polish  
3        Drama     May 27, 2021       95         6.7  English  
4       Horror     May 26, 2021      117         5.2     Thai

In [122]:
df_concat.loc[df_concat["type"] == "Movie"]

show_id   type                                             title  \
0        s1  Movie                              Dick Johnson Is Dead   
1      s142  Movie                                        Extraction   
2      s625  Movie                                        Prime Time   
3      s835  Movie                                      Blue Miracle   
4      s837  Movie                                         Ghost Lab   
..      ...    ...                                               ...   
508   s5902  Movie                                               Tig   
509   s5907  Movie                       What Happened, Miss Simone?   
510   s5911  Movie                                  Hot Girls Wanted   
511   s5913  Movie  The Other One: The Long Strange Trip of Bob Weir   
512   s5917  Movie                                        My Own Man   

                          director  \
0                  Kirsten Johnson   
1                 Steven C. Miller   
2                     Jakub Piątek   
3                   Julio Quintana   
4              Paween Purijitpanya   
..                             ...   
508  Kristina Goolsby, Ashley York   
509                     Liz Garbus   
510       Jill Bauer, Ronna Gradus   
511                    Mike Fleiss   
512                David Sampliner   

                                                  cast  \
0                                                  NaN   
1    Bruce Willis, Kellan Lutz, Gina Carano, D.B. S...   
2    Bartosz Bielenia, Magdalena Popławska, Andrzej...   
3    Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...   
4    Thanapob Leeratanakachorn, Paris Intarakomalya...   
..                                                 ...   
508                                         Tig Notaro   
509                                                NaN   
510                                                NaN   
511                                           Bob Weir   
512                                    David Sampliner   

                                   country          date_added  release_year  \
0                            United States  September 25, 2021          2020   
1    United States, United Kingdom, Canada   September 1, 2021          2015   
2                                      NaN       June 30, 2021          2021   
3                            United States        May 27, 2021          2021   
4                                 Thailand        May 27, 2021          2021   
..                                     ...                 ...           ...   
508                          United States       July 17, 2015          2015   
509                          United States       June 26, 2015          2015   
510                          United States        May 29, 2015          2015   
511                          United States        May 22, 2015          2015   
512                          United States       March 6, 2015          2015   

    rating duration                                          listed_in  \
0    PG-13   90 min                                      Documentaries   
1        R   82 min                                 Action & Adventure   
2    TV-MA      NaN            Dramas, International Movies, Thrillers   
3    TV-PG      NaN  Children & Family Movies, Dramas, Faith & Spir...   
4    TV-MA  118 min     Horror Movies, International Movies, Thrillers   
..     ...      ...                                                ...   
508  TV-14      NaN                     Documentaries, Stand-Up Comedy   
509  TV-14      NaN                    Documentaries, Music & Musicals   
510  TV-MA   83 min                                      Documentaries   
511  TV-14   84 min                    Documentaries, Music & Musicals   
512  TV-MA   82 min                                      Documentaries   

                                           description  \
0    As her father nears the end of his life, filmm...   
1    When a retired

In [123]:
df_concat.loc[df_concat["type"] == "TV Show",["title", "duration", "type"]]

Empty DataFrame
Columns: [title, duration, type]
Index: []

In [124]:
df_concat.loc[df_concat["country"] == "United States"]

show_id   type                                             title  \
0        s1  Movie                              Dick Johnson Is Dead   
3      s835  Movie                                      Blue Miracle   
7      s854  Movie                                  Army of the Dead   
11     s890  Movie                           The Woman in the Window   
14     s911  Movie                                           Monster   
..      ...    ...                                               ...   
508   s5902  Movie                                               Tig   
509   s5907  Movie                       What Happened, Miss Simone?   
510   s5911  Movie                                  Hot Girls Wanted   
511   s5913  Movie  The Other One: The Long Strange Trip of Bob Weir   
512   s5917  Movie                                        My Own Man   

                          director  \
0                  Kirsten Johnson   
3                   Julio Quintana   
7                      Zack Snyder   
11                      Joe Wright   
14                 Anthony Mandler   
..                             ...   
508  Kristina Goolsby, Ashley York   
509                     Liz Garbus   
510       Jill Bauer, Ronna Gradus   
511                    Mike Fleiss   
512                David Sampliner   

                                                  cast        country  \
0                                                  NaN  United States   
3    Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...  United States   
7    Dave Bautista, Ella Purnell, Omari Hardwick, G...  United States   
11   Amy Adams, Gary Oldman, Anthony Mackie, Fred H...  United States   
14   Kelvin Harrison Jr., Jennifer Hudson, Jeffrey ...  United States   
..                                                 ...            ...   
508                                         Tig Notaro  United States   
509                                                NaN  United States   
510                                                NaN  United States   
511                                           Bob Weir  United States   
512                                    David Sampliner  United States   

             date_added  release_year rating duration  \
0    September 25, 2021          2020  PG-13   90 min   
3          May 27, 2021          2021  TV-PG      NaN   
7          May 21, 2021          2021      R  148 min   
11         May 14, 2021          2021      R      NaN   
14          May 7, 2021          2021  TV-MA      NaN   
..                  ...           ...    ...      ...   
508       July 17, 2015          2015  TV-14      NaN   
509       June 26, 2015          2015  TV-14      NaN   
510        May 29, 2015          2015  TV-MA   83 min   
511        May 22, 2015          2015  TV-14   84 min   
512       March 6, 2015          2015  TV-MA   82 min   

                                             listed_in  \
0                                        Documentaries   
3    Children & Family Movies, Dramas, Faith & Spir...   
7                    Action & Adventure, Horror Movies   
11                                   Dramas, Thrillers   
14                                              Dramas   
..                                                 ...   
508                     Documentaries, Stand-Up Comedy   
509                    Documentaries, Music & Musicals   
510                                      Documentaries   
511                    Documentaries, Music & Musicals   
512                                      Documentaries   

                                           description  \
0    As her father nears the end of his life, filmm...   
3    To save their cash-strapped orphanage, a guard...   
7    After a zombie outbreak in Las Vegas, a group ...   
11   Confined to her home by agoraphobia, a psychol...   
14   A talented teen implicated in a robbery-turned...   
..                                                 ...   
508  Comedian Tig

In [125]:
df_concat.loc[df_concat["type"] == "Movie",["title", "director"]]

title  \
0                                Dick Johnson Is Dead   
1                                          Extraction   
2                                          Prime Time   
3                                        Blue Miracle   
4                                           Ghost Lab   
..                                                ...   
508                                               Tig   
509                       What Happened, Miss Simone?   
510                                  Hot Girls Wanted   
511  The Other One: The Long Strange Trip of Bob Weir   
512                                        My Own Man   

                          director  
0                  Kirsten Johnson  
1                 Steven C. Miller  
2                     Jakub Piątek  
3                   Julio Quintana  
4              Paween Purijitpanya  
..                             ...  
508  Kristina Goolsby, Ashley York  
509                     Liz Garbus  
510       Jill Bauer, Ronna Gradus  
511                    Mike Fleiss  
512                David Sampliner  

[513 rows x 2 columns]

In [126]:
df_concat.loc[df_concat["type"] == "Movie", ["title", "director"]]

title  \
0                                Dick Johnson Is Dead   
1                                          Extraction   
2                                          Prime Time   
3                                        Blue Miracle   
4                                           Ghost Lab   
..                                                ...   
508                                               Tig   
509                       What Happened, Miss Simone?   
510                                  Hot Girls Wanted   
511  The Other One: The Long Strange Trip of Bob Weir   
512                                        My Own Man   

                          director  
0                  Kirsten Johnson  
1                 Steven C. Miller  
2                     Jakub Piątek  
3                   Julio Quintana  
4              Paween Purijitpanya  
..                             ...  
508  Kristina Goolsby, Ashley York  
509                     Liz Garbus  
510       Jill Bauer, Ronna Gradus  
511                    Mike Fleiss  
512                David Sampliner  

[513 rows x 2 columns]

In [129]:
df_concat.loc[df_concat["release_year"] == 2018, ["title", "listed_in"]]

title  \
251             Little Miss Sumo   
320           Nappily Ever After   
324     Period. End of Sentence.   
331                         Soni   
334                    Lionheart   
..                           ...   
413      The Cloverfield Paradox   
414  A Futile and Stupid Gesture   
415                 Step Sisters   
416               The Open House   
417               The Polka King   

                                             listed_in  
251  Documentaries, International Movies, Sports Mo...  
320                  Comedies, Dramas, Romantic Movies  
324                                      Documentaries  
331   Dramas, Independent Movies, International Movies  
334               Comedies, Dramas, Independent Movies  
..                                                 ...  
413               Action & Adventure, Sci-Fi & Fantasy  
414                                           Comedies  
415                                           Comedies  
416                           Horror Movies, Thrillers  
417                       Comedies, Independent Movies  

[87 rows x 2 columns]

In [132]:
df_concat.loc[df_concat["director"] == "", ["title"]]

Empty DataFrame
Columns: [title]
Index: []